File: app_manager/appmanager/src/main/java/app/editors/manager/ui/fragments/main/DocsMyFragment.java
package app.editors.manager.ui.fragments.main;

import android.os.Bundle;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import app.editors.manager.managers.providers.CloudFileProvider;
import app.editors.manager.ui.activities.main.MainActivity;

public class DocsMyFragment extends DocsCloudFragment {

    public static DocsMyFragment newInstance() {
        return new DocsMyFragment();
    }

    public static final String ID = CloudFileProvider.Section.My.getPath();

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        init();
    }

    @Override
    protected boolean onSwipeRefresh() {
        if (!super.onSwipeRefresh() && mCloudPresenter != null) {
            mCloudPresenter.getItemsById(ID);
            return true;
        }

        return false;
    }

    @Override
    public void onScrollPage() {
        super.onScrollPage();
        if (mCloudPresenter.getStack() == null) {
            mCloudPresenter.getItemsById(ID);
        }
    }


    @Override
    public void onStateEmptyBackStack() {
        super.onStateEmptyBackStack();
        if (mSwipeRefresh != null) {
            mSwipeRefresh.setRefreshing(true);
        }
        mCloudPresenter.getItemsById(ID);
    }

    private void init() {
        mExplorerAdapter.setSectionMy(true);
        mCloudPresenter.checkBackStack();
        getArgs();
    }
}

File: app_manager/appmanager/src/main/java/app/editors/manager/ui/fragments/main/DocsMyFragment.kt
package app.editors.manager.ui.fragments.main

import android.os.Bundle
import android.view.View
import app.documents.core.network.ApiContract
import app.editors.manager.managers.providers.CloudFileProvider

class DocsMyFragment : DocsCloudFragment() {

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        init()
    }

    override fun onSwipeRefresh(): Boolean {
        if (!super.onSwipeRefresh()) {
            cloudPresenter.getItemsById(ID)
            return true
        }
        return false
    }

    override fun onScrollPage() {
        super.onScrollPage()
        if (cloudPresenter.stack == null) {
            cloudPresenter.getItemsById(ID)
        }
    }

    override val section: Int
        get() = ApiContract.SectionType.CLOUD_USER

    override fun onStateEmptyBackStack() {
        super.onStateEmptyBackStack()
        swipeRefreshLayout?.isRefreshing = true
        cloudPresenter.getItemsById(ID)
    }

    override fun onUpdateItemFavorites() {
        explorerAdapter?.updateItem(cloudPresenter.itemClicked)
    }

    private fun init() {
        explorerAdapter?.isSectionMy = true
        cloudPresenter.checkBackStack()
    }

    companion object {
        val ID = CloudFileProvider.Section.My.path

        fun newInstance(stringAccount: String): DocsMyFragment {
            return DocsMyFragment().apply {
                arguments = Bundle(1).apply {
                    putString(KEY_ACCOUNT, stringAccount)
                }
            }
        }
    }
}